feat(clerk-js,types): Do not require current password if Reverifaction enabled (#5284)
clerk/javascript: .changeset/heavy-hornets-yawn.md

---
'@clerk/clerk-js': patch
'@clerk/types': patch
'@clerk/clerk-expo': patch
---

> In case Reverification is enabled, then we don't require the user's current password. This is in order to improve the UX, as the user has already verified themselves
> using Reverification, so there is no point to maintain a two level verification in case they would like to change their password. Also, Reverification is a stronger
> verification factor, as it includes strategies such as email code.

Reviewer: Adding the first line as a header, so that in the changeset the rest of the information will be added under this header.

```suggestion
Improve the UX on Reverification by not requiring the user's current password. 

The user has already verified themselves using Reverification, so there is no point to maintain a two level verification in case they would like to change their password. Also, Reverification is a stronger verification factor, as it includes strategies such as email code.
```
